### ISSUE TITLE
Cant cat in Entargling Roots and Hamstring spells (BB #665)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#665]
This issue was migrated from bitbucket.
**Original Reporter:** smoldar
**Original Date:** 23.08.2014 08:58:55 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/665

<hr>

hamstring and roots from druid creating stuns for no move... but in these stuns you cant cast any spells... its look like as silence now :D

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 23.08.2014 09:41:02 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/665/#comment-11946391

<hr>

Spells or wands

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 23.08.2014 12:15:41 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/665/#comment-11947308

<hr>

I can't confirm this one.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 24.08.2014 07:43:05 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/665/#comment-11953985

<hr>

Can't reproduce. 
Tested personally with Hamstring and Fireball against an NPC.
Revision 2271

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 25.08.2014 07:41:35 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/665/#comment-11966001

<hr>

I can't reproduce too... but 2 friends spamed me with this... i think it is invalid.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 27.08.2014 12:27:45 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/665/#comment-12015639

<hr>

This bug exist. But you must get these immobilize spells or stuns while you moving. If you stay ... you dont get this bug.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 27.08.2014 12:29:22 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/665/#comment-12015665

<hr>

You will see message something about ,,You cant do it while moving"  

I think these rows...

```
#!c++

+    // don't allow channeled spells / spells with cast time to be casted while moving
Add a comment to this line
+    // (even if they are interrupted on moving, spells with almost immediate effect get to have their effect processed before movement interrupter kicks in)
+    if ((IsChanneledSpell(m_spellInfo) || m_casttime)
+        && m_caster->GetTypeId() == TYPEID_PLAYER && m_caster->isMoving()
+        && m_spellInfo->InterruptFlags & SPELL_INTERRUPT_FLAG_MOVEMENT)
+    {
+        SendCastResult(SPELL_FAILED_MOVING);
+        finish(false);
+        return;
+    }
+
```

Created by commit [5b26cb7](https://bitbucket.org/oregon/oregoncore/commits/5b26cb73890ad3a551569d54091bf5c5de4c4dae)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 27.08.2014 12:40:15 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/665/#comment-12015846

<hr>

Could this be due to movement flags for moving not being removed in roots
etc

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 27.08.2014 12:47:55 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/665/#comment-12016032

<hr>

Yes it can be by not removed flags.

This bug exist on Hamstring, Roots and Feral Druid Charge i think these. If it exist on other spells i will send more info.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 27.08.2014 16:49:09 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/665/#comment-12021750

<hr>

Fix issue #665

Unit is not actually moving it has MOVEFLAG_ROOT set.

→ <<cset 2989ee9af074>>
